phys/module_physics_init.F

```diff
@@ -938,7 +938,22 @@ SUBROUTINE phy_init ( id, config_flags, DT, restart, zfull, zhalf,     &
       has_reqs = 0
    end if
 
-   ENDIF
+! If using FARMS, we need both effective radii and the mass
+
+   IF ( config_flags%swint_opt .eq. swintopt2 ) THEN
+      IF ( ( has_reqc .EQ. 1 ) .AND. &
+           ( has_reqi .EQ. 1 ) .AND. &
+           ( has_reqs .EQ. 1 ) .AND. &
+           ( f_qc            ) .AND. &
+           ( f_qi            ) .AND. &
+           ( f_qs            ) ) THEN
+         !  everything is A-OK for FARMS
+      ELSE
+         CALL wrf_error_fatal ('--- ERROR: FARMS requires a different MP scheme')
```

Comment:
@davegill Could we add (swint_opt=2) after FARMS?

Reply:
Wei,
Good suggestion. I put that in.